### PR TITLE
Disable extended tests if mongodb version is lower than 2.6

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @rmg

### CODE REVIEW

File: test/mongodb.test.js [hunk 1]
```diff
@@ -499,149 +499,162 @@ describe('mongodb', function () {
...
-            should.not.exist(err3);
-
-            User.updateAll({name: 'Simon'}, {name: 'Alex'}, function(err, updatedusers) {
-              should.not.exist(err);
-              updatedusers.should.be.equal(1);
+    describe('extended operators', function() {
+      if (process.env.MONGODB_VERSION) {
+        var semver = require('semver');
+        // semver.lt doesn't accept 2.4 (2.4.0 is required)
+        if (semver.satisfies('2.6.0', '>' + process.env.MONGODB_VERSION)) {
```
Comment (rmg) [inline test/mongodb.test.js]: I think you mean:
```js
if (!semver.satisfies(process.env.MONGODB_VERSION, '>=2.6.0')) {
```

**Comment by rmg** [comment]
Rather than putting the test directly in the block.. how about:
```js
var describeMongo26 = describe;
if (process.env.MONGODB_VERSION && !semver.satisfies(process.env.MONGODB_VERSION, '>=2.6.0')) {
  describeMongo26 = describe.skip;
}
describeMongo26('extended operators', function() {
  // ... describe block is run or marked as skipped
});
```

**Comment by rmg** [comment]
Or you could do something similar with `it` and `it.skip`, which might remove the need to wrap everything in a `describe` block and change all that whitespace.

File: test/mongodb.test.js (hunk 1 above)
Comment (raymondfeng) [inline test/mongodb.test.js]: I had to use `semver.satisfies('2.6.0', '>' + process.env.MONGODB_VERSION)` as `!semver.satisfies(process.env.MONGODB_VERSION, '>=2.6.0')` throws if process.env.MONGODB_VERSION is `2.4`

**Comment by raymondfeng** [comment]
@rmg PR updated.

File: test/mongodb.test.js (hunk 1 above)
Comment (rmg) [inline test/mongodb.test.js]: I saw the comment, but it wasn't clear that it was related to reversing the arguments. I understand now.

**Comment by rmg** [comment]
I'm still not a fan of how the range check is done because it makes the code very hard to follow, but I also don't want to block this PR based on differences in coding style, so LGTM.

FWIW, the actual version strings being reported are `v2.6.7' and `v2.4.9`, so they won't cause the invalid version specifier problem.
